tests/unit/test_parameters.py

```diff
@@ -9,15 +9,68 @@
 from databricks.sql.thrift_api.TCLIService.ttypes import (
     TSparkParameter,
     TSparkParameterValue,
+    TSessionHandle,
+    TOpenSessionResp,
 )
 from databricks.sql.utils import DbSqlParameter, DbSqlType
 import pytest
 
-from decimal import Decimal
+from databricks.sql.thrift_backend import ThriftBackend
+from databricks.sql.thrift_api.TCLIService import ttypes
 
+from decimal import Decimal
+from databricks.sql.client import Connection
 from typing import List
 
 
+class TestSessionHandleChecks(object):
+    def test_get_session_handle(self):
+        assert (
+            Connection.get_protocol_version(
+                TOpenSessionResp(
+                    serverProtocolVersion=ttypes.TProtocolVersion.SPARK_CLI_SERVICE_PROTOCOL_V7,
+                    sessionHandle=TSessionHandle(1, None),
+                )
+            )
+            == ttypes.TProtocolVersion.SPARK_CLI_SERVICE_PROTOCOL_V7
```

Comment:
Per my comment on the definition of `get_protocol_version`, the test names or comments should explain what we're testing for here. These assertions verify that `get_protocol_version` falls back to the `openSessionResp.serverProtocolVersion` if it's not present in `sessionHandle`. We should have some indication as to _why_.